src/auth/src/credentials/mds.rs

```diff
@@ -352,6 +502,53 @@ mod test {
         assert!(result.is_err());
     }
 
+    #[tokio::test]
+    async fn get_headers_success_with_quota_project() {
+        let scopes = vec!["scope1".to_string(), "scope2".to_string()];
+        let response = MDSTokenResponse {
+            access_token: "test-access-token".to_string(),
+            expires_in: Some(3600),
+            token_type: "test-token-type".to_string(),
+        };
+        let response_body = serde_json::to_value(&response).unwrap();
+
+        let (endpoint, _server) = start(HashMap::from([(
+            MDS_TOKEN_URI.to_string(),
+            (
+                StatusCode::OK,
+                response_body,
+                TokenQueryParams {
+                    scopes: Some(scopes.join(",")),
+                    recursive: None,
+                },
+            ),
+        )]))
+        .await;
+
+        let mdsc = Builder::default()
+            .scopes(scopes)
```

Comment:
You should test the `IntoIterator` thing at least once, maybe here write:

```suggestion
            .scopes(["scope1", "scope2"])
```